src/data/finset/locally_finite.lean

```diff
@@ -515,4 +517,23 @@ lemma image_add_right_Ioo (a b c : α) : (Ioo a b).image (+ c) = Ioo (a + c) (b
 by { simp_rw add_comm _ c, exact image_add_left_Ioo a b c }
 
 end ordered_cancel_add_comm_monoid
+
+@[to_additive] lemma prod_Ioi_mul_prod_Ioi_mul_eq_prod_off_diag [fintype ι] [linear_order ι]
+  [locally_finite_order_top ι] [locally_finite_order_bot ι] [comm_monoid α] (f : ι → ι → α) :
+  ∏ i, ∏ j in Ioi i, f j i * f i j = ∏ i, ∏ j in univ.filter (λ j, i ≠ j), f j i :=
+begin
+  have Ioiio : ∀ i : ι, univ.filter (λ j, i ≠ j) = Ioi i ∪ Iio i,
```

Comment:
This would be good to have as a lemma, in the form
```suggestion
  have Ioiio : ∀ i : ι, univ.filter (λ j, i ≠ j) = (Ioi i).disj_union (Iio i),
```